src/main/java/org/isf/medicalstockward/manager/MovWardBrowserManager.java

```diff
@@ -295,4 +296,54 @@ public int compare(MovementForPrint o1, MovementForPrint o2) {
 		}
 	}
 
+	/* * 
+	 * Delete the MovementWard for the specified criteria.
+	 *
+	 * @param movWardToDelete - the movement ward
+	 * @throws OHServiceException if an error occurs retrieving the medical.
+	 */
+	@Transactional(rollbackFor = OHServiceException.class)
+	public void deleteLastMovementWard(MovementWard movWardToDelete) throws OHServiceException {
+		MovementWard lastMovementWard = ioOperations.getLastMovementWard(movWardToDelete.getWard());
+		if (lastMovementWard.getCode() != movWardToDelete.getCode()) {
+			throw new OHDataValidationException(
+							new OHExceptionMessage(MessageBundle.getMessage("angal.medicalstock.onlythelastmovementcanbedeleted.msg")));
+		}
+		if (movWardToDelete.getWardTo() != null) {
+			MovementWard lastMovInWardTo = ioOperations.getLastMovementWard(movWardToDelete.getWardTo());
+			if (lastMovInWardTo.getWard().getCode().equals(lastMovementWard.getWardTo().getCode())) {
```

Comment:
I think this check is obvious because at this point `lastMovementWard` = `movWardToDelete`, so also `lastMovementWard.getWardTo()` = `movWardToDelete.getWardTo()` and so `lastMovInWardTo.getWard()`

Reply:
I agree with you but I think it is just the precaution that I have take to check that. If you agree to remove it I can remove.

Reply:
I mean, if the check is `false`, then the error is `notpossibletodeletethismovementthemedicalhasbeenusedafterbeenreceivedinward`, but how you can say so?

Reply:
the point is not to remove the condition, but to fix it? otherwise in which case you'd have the `notpossibletodeletethismovementthemedicalhasbeenusedafterbeenreceivedinward` error? Deletion is always allowed?

Reply:
When deleting a transfer movement from ward to ward, you have fisrt to check if the medical has been used in the destination ward and if so, deny the deletion, otherwise you have to restore quantities in both origin and destination ward and them delete the transfer movement

Reply:
okay